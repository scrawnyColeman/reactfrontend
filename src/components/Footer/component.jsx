import React from 'react';
import styled from 'styled-components';
import { colours } from '../../constants/styles';

const StyledFooter = styled.div`
    position: fixed;
    bottom: 0;
    left: 0;
    height: 30px;
    padding: 12px 0;
    width: 100%;
    color: ${colours.secondary};
`;

const Footer = () => {
    const date = new Date();
    return <StyledFooter>{`Â© Njoy Learning ${date.getFullYear()}. All rights reserved`}</StyledFooter>;
};

export default Footer;
