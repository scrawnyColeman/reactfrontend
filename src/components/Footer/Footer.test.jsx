import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import Footer from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('Footer component', () => {
    const RealDate = Date;

    function mockDate(isoDate) {
        global.Date = class {
            constructor() {
                return new RealDate(isoDate);
            }
        };
    }
    afterAll(() => {
        global.Date = RealDate;
    });

    test('component matches snapshot', () => {
        mockDate('2020-06-03T12:34:56z');
        const wrapper = shallow(<Footer />);
        expect(wrapper).toMatchSnapshot();
    });
});
