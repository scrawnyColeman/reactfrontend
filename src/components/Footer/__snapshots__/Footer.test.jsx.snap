// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component component matches snapshot 1`] = `
<styled.div>
  Â© Njoy Learning 2020. All rights reserved
</styled.div>
`;
